context.go

```diff
@@ -25,6 +26,7 @@ func contextFromOptions(merged *WithOption) *context {
 		cacheOnly:           *merged.CacheOnly,
 		cachePath:           getCachePath(),
 		disableNetworkFetch: *merged.DisableNetworkFetch,
+		directPath:          *merged.DirectPath,
```

Comment:
How about we call this just `Path` instead of `DirectPath`? since it's an option, if `Path` is set, it overrides any of the auto-discovery features.